### DISCUSSION TITLE
Bind address -D | socks host

### DISCUSSION THREAD

**Discussion by NativoDigital** [isontheline/pro.webssh.net#867] [Q&A]
Hi, thanks for this great app and congratulations for the work.

I’m trying since some days ago to get working the VPN-Over-SSH function in combination with -D (bind_address) option because, as I understand, for tunneling using local socks as I usually do in Putty with Tunnels -> Dynamic in Windows or in a Linux shell with a command like this -> ssh -4fN -D 45666 user@server.com I need this config:

![image](https://user-images.githubusercontent.com/16826447/231310339-be9d394a-48a8-4ff0-ad2f-33bbc93bef54.jpeg)

This is the simple configuration that I’m using following the instructions and letting WebSSH adds an VPN connection to the ios config.

Then, the vpn appears connected but when I visit some site with, for example Firefox, the ip is my home ip or my “real” ip.

**Comment by isontheline** [comment]
Hi @NativoDigital 👋

Thank you so much for reaching me out and for your support about WebSSH 🙏

Yet VPN-Over-SSH doesn't forward your device streams to your tunnel. It only forwards connections inside WebSSH itself.

Issue #104 is needed to enable connections outside WebSSH.

I will try to make a look at it as soon as I can.

**Reply by NativoDigital** [reply]
Thanks for your quick response!

Sorry I thought that was already implemented with VPN-Over-SSH. Usually a socks proxy is intended to be used by one or some applications at a time, In this case firefox should have in the configuration the option for connecting through a local SOCKS5 proxy. But it hasn’t got, as it has in the desktop versión. Telegram does have the option.

So i’m not sure If it is more convenient to send the whole mobile data via our remote server or by the application because maybe it has no proxy options, or even if all that is possible :)

**Reply by isontheline** [reply]
You're right only some apps allow to set SOCKS proxy settings 😢

I'm also searching for a way to implement a "Safari" like inside WebSSH that will use embedded SOCKS proxy. No luck, yet.
